### PR TITLE
Making numpy_groupies an extra requirement

### PR DESCRIPTION
**Pull request by roxyboy**
I moved the call `import numpy_groupies` only to when `isotropic` functions are called.

### CODE REVIEW

File: xrft/xrft.py [hunk 1]
```diff
@@ -782,6 +781,9 @@ def _binned_agg(
     dtype,
 ) -> np.ndarray:
     """NumPy helper function for aggregating over bins."""
+
+    import numpy_groupies
```
Comment (rabernat) [inline xrft/xrft.py:785]:
```suggestion
    try:
        import numpy_groupies
    except ImportError:
        raise ImportError("This function requires the `numpy_groupies` package to be installed. Please install it with pip or conda.")
```

**Review by rabernat — COMMENTED** [review]
It would be nice to have a more verbose error message.

File: xrft/xrft.py (hunk 1 above)
Comment (roxyboy) [reply xrft/xrft.py]: I added the import error. I've never made a test to catch such error but do we need one?

Comment (rabernat) [reply xrft/xrft.py]: I think it helps the user understand better. Just imagine you are a python n00b and you try this function and you get a random import error for a package you have never heard of!

Comment (roxyboy) [reply xrft/xrft.py]: I agree with your point. I'm just wondering if I'd need to add another test in `test_xrft.py` to check for this `ImportError`...

Comment (rabernat) [reply xrft/xrft.py]: Not necessary IMO.

Comment (roxyboy) [reply xrft/xrft.py]: Ok cool. So we can merge this?

**Pull request merged**